zingolib/src/wallet/notes/transparent.rs

```diff
@@ -53,6 +54,47 @@ impl OutputInterface for TransparentOutput {
     fn pending_spent_mut(&mut self) -> &mut Option<(TxId, u32)> {
         &mut self.unconfirmed_spent
     }
+
+    fn transaction_record_to_outputs_vec(transaction_record: &TransactionRecord) -> Vec<&Self> {
+        let mut vecref = vec![];
+        for output in &transaction_record.transparent_outputs {
+            vecref.push(output);
+        }
+        vecref
+    }
+    fn transaction_record_to_outputs_vec_query(
+        transaction_record: &TransactionRecord,
+        spend_status_query: OutputSpendStatusQuery,
+    ) -> Vec<&Self> {
+        let mut vecref = vec![];
+        for output in &transaction_record.transparent_outputs {
+            if output.spend_status_query(spend_status_query) {
+                vecref.push(output);
+            }
+        }
+        vecref
```

Comment:
Not as many fewer lines but still likely easier to follow
```suggestion
        transaction_record
            .sapling_notes
            .iter()
            .filter(|output| output.spend_status_query(spend_status_query))
            .collect()

```